query.go

```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
 	frame.Meta.ExecutedQueryString = query.RawSQL
 	frame.Meta.PreferredVisualization = data.VisTypeGraph
 
-	if query.Format == FormatOptionTable {
-		frame.Meta.PreferredVisualization = data.VisTypeTable
-		return data.Frames{frame}, nil
+	count, err := frame.RowLen()
+	if err != nil {
+		return nil, err
+	}
+	if count == 0 {
+		return nil, ErrorNoResults
 	}
 
-	if query.Format == FormatOptionLogs {
+	switch query.Format {
+	case FormatOptionTable:
+		frame.Meta.PreferredVisualization = data.VisTypeTable
+		return data.Frames{frame}, nil
+	case FormatOptionLogs:
 		frame.Meta.PreferredVisualization = data.VisTypeLogs
 		return data.Frames{frame}, nil
-	}
-
-	if query.Format == FormatOptionTrace {
+	case FormatOptionTrace:
 		frame.Meta.PreferredVisualization = data.VisTypeTrace
 		return data.Frames{frame}, nil
-	}
+	case FormatOptionMulti:
+		if frame.TimeSeriesSchema().Type == data.TimeSeriesTypeLong {
 
-	count, err := frame.RowLen()
+			err = fixFrameForLongToMulti(frame)
+			if err != nil {
+				return nil, err
+			}
 
-	if err != nil {
-		return nil, err
+			frames, err := timeseries.LongToMulti(&timeseries.LongFrame{frame})
+			if err != nil {
+				return nil, err
+			}
+			return frames.Frames(), nil
+		}
```

Comment:
Should it be an error to specifiy `FormatOptionMulti` with a different type here? (I don't expect so, mostly asking for my own edification)

Reply:
No, I don't think it's an error. (Though we're not supporting wide to multi, I suspect that's not a use case we expect)